Use DistSQL add resource have unexpected behavior

## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?

`e6328cb989cfb789189227892cb892740095d2d6`

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy

### Expected behavior
add resource have error .so `show database resources` should't have result
### Actual behavior
add resource have error . excute `show database resources` have result
<img width="1855" alt="image" src="https://user-images.githubusercontent.com/86462784/180726407-73712d41-05f5-480a-a8f0-73de8e5dd525.png">

### Reason analyze (If you can)

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
1. use server.yaml start proxy
server.yaml
```
mode:
  type: Cluster
  repository:
    type: ZooKeeper
    props:
      namespace: 0715distsql
      server-lists: 192.168.10.21:2181
      retryIntervalMilliseconds: 500
      timeToLiveSeconds: 60
      maxRetries: 3
      operationTimeoutMilliseconds: 500
  overwrite: true

rules:
- !AUTHORITY
  users:
    - root@%:root
    - sharding@:sharding
  provider:
    type: ALL_PERMITTED
- !SQL_PARSER
  sqlCommentParseEnabled: true
  sqlStatementCache:
    initialCapacity: 2000
    maximumSize: 65535
  parseTreeCache:
    initialCapacity: 128
    maximumSize: 1024

props:
  proxy-frontend-database-protocol-type: "PostgreSQL"
```
2. psql -U root  -h 127.0.0.1 -p 3333(proxy  port) -d postgres
3. execute sql
```
ADD RESOURCE resource_0 (
 URL="jdbc:postgresql://192.168.10.21:5432/postgres",
 USER=gaussdb,
 PASSWORD=passwd
);
```
```
show database resources;
```
### Example codes for reproduce this issue (such as a github link).

<img width="1681" alt="image" src="https://user-images.githubusercontent.com/86462784/179918206-16624db1-ef2d-46a2-b285-59b784d5327d.png">

### stdout.log

```
[ERROR] 2022-07-20 14:58:22.059 [Connection-2-ThreadExecutor] o.a.s.p.b.t.d.r.r.AddResourceBackendHandler - Add resource failed
java.lang.reflect.InvocationTargetException: null
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.shardingsphere.transaction.xa.jta.datasource.swapper.DataSourceSwapper.setProperties(DataSourceSwapper.java:118)
        at org.apache.shardingsphere.transaction.xa.jta.datasource.swapper.DataSourceSwapper.swap(DataSourceSwapper.java:57)
        at org.apache.shardingsphere.transaction.xa.jta.datasource.XADataSourceFactory.build(XADataSourceFactory.java:43)
        at org.apache.shardingsphere.transaction.xa.jta.datasource.XATransactionDataSource.<init>(XATransactionDataSource.java:65)
        at org.apache.shardingsphere.transaction.xa.XAShardingSphereTransactionManager.newXATransactionDataSource(XAShardingSphereTransactionManager.java:62)
        at org.apache.shardingsphere.transaction.xa.XAShardingSphereTransactionManager.lambda$init$0(XAShardingSphereTransactionManager.java:56)
        at java.util.ArrayList.forEach(ArrayList.java:1259)
        at org.apache.shardingsphere.transaction.xa.XAShardingSphereTransactionManager.init(XAShardingSphereTransactionManager.java:56)
        at org.apache.shardingsphere.transaction.ShardingSphereTransactionManagerEngine.lambda$init$0(ShardingSphereTransactionManagerEngine.java:66)
        at java.util.Map.forEach(Map.java:630)
        at org.apache.shardingsphere.transaction.ShardingSphereTransactionManagerEngine.init(ShardingSphereTransactionManagerEngine.java:66)
        at org.apache.shardingsphere.transaction.rule.TransactionRule.createTransactionManagerEngine(TransactionRule.java:76)
        at org.apache.shardingsphere.transaction.rule.TransactionRule.rebuildEngine(TransactionRule.java:110)
        at org.apache.shardingsphere.transaction.rule.TransactionRule.addResource(TransactionRule.java:87)
        at org.apache.shardingsphere.mode.manager.ContextManager.lambda$updateResources$3(ContextManager.java:208)
        at java.lang.Iterable.forEach(Iterable.java:75)
        at org.apache.shardingsphere.mode.manager.ContextManager.updateResources(ContextManager.java:208)
        at org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.AddResourceBackendHandler.execute(AddResourceBackendHandler.java:63)
        at org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.AddResourceBackendHandler.execute(AddResourceBackendHandler.java:44)
        at org.apache.shardingsphere.proxy.backend.text.DatabaseRequiredBackendHandler.execute(DatabaseRequiredBackendHandler.java:51)
        at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.simple.PostgreSQLComQueryExecutor.execute(PostgreSQLComQueryExecutor.java:79)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:108)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:78)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: URL invalid jdbc:opengauss://[ip:5432/postgres](http://ip:5432/postgres)
        at org.postgresql.ds.common.BaseDataSource.setUrl(BaseDataSource.java:1313)
        ... 30 common frames omitted
[ERROR] 2022-07-20 14:58:22.060 [Connection-2-ThreadExecutor] o.a.s.p.f.c.CommandExecutorTask - Exception occur:
org.apache.shardingsphere.infra.distsql.exception.resource.InvalidResourcesException: Can not process invalid resources, error messages are: [null].
        at org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.AddResourceBackendHandler.execute(AddResourceBackendHandler.java:68)
        at org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.AddResourceBackendHandler.execute(AddResourceBackendHandler.java:44)
        at org.apache.shardingsphere.proxy.backend.text.DatabaseRequiredBackendHandler.execute(DatabaseRequiredBackendHandler.java:51)
        at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.simple.PostgreSQLComQueryExecutor.execute(PostgreSQLComQueryExecutor.java:79)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:108)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:78)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

### server.yaml
```
mode:
  type: Cluster
  repository:
    type: ZooKeeper
    props:
      namespace: 0720distsql
      server-lists: 192.168.10.21:2181
      retryIntervalMilliseconds: 500
      timeToLiveSeconds: 60
      maxRetries: 3
      operationTimeoutMilliseconds: 500
  overwrite: true

rules:
- !AUTHORITY
  users:
    - root@%:root
    - sharding@:sharding
  provider:
    type: ALL_PERMITTED
- !SQL_PARSER
  sqlCommentParseEnabled: true
  sqlStatementCache:
    initialCapacity: 2000
    maximumSize: 65535
  parseTreeCache:
    initialCapacity: 128
    maximumSize: 1024

props:
  proxy-frontend-database-protocol-type: "PostgreSQL"
```

--- a/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesValidator.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesValidator.java
@@ -17,6 +17,8 @@
 
 package org.apache.shardingsphere.infra.datasource.props;
 
+import lombok.RequiredArgsConstructor;
+import org.apache.shardingsphere.infra.database.type.DatabaseType;
 import org.apache.shardingsphere.infra.datasource.pool.creator.DataSourcePoolCreator;
 import org.apache.shardingsphere.infra.datasource.pool.destroyer.DataSourcePoolDestroyer;
 import org.apache.shardingsphere.infra.distsql.exception.resource.InvalidResourcesException;
@@ -31,7 +33,10 @@ import java.util.Map.Entry;
 /**
  * Data source properties validator.
  */
+@RequiredArgsConstructor
 public final class DataSourcePropertiesValidator {
+    
+    private final DatabaseType databaseType;
     
     /**
      * Validate data source properties map.
@@ -70,6 +75,8 @@ public final class DataSourcePropertiesValidator {
     }
     
     private void checkFailFast(final DataSource dataSource) throws SQLException {
-        dataSource.getConnection();
+        if (!dataSource.getConnection().getMetaData().getDatabaseProductName().equals(databaseType.getType().toLowerCase())) {
+            throw new SQLException("Protocol mismatch for datasource.");
+        }
     }
 }

--- a/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesValidator.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesValidator.java
@@ -17,7 +17,6 @@
 
 package org.apache.shardingsphere.infra.datasource.props;
 
-import lombok.RequiredArgsConstructor;
 import org.apache.shardingsphere.infra.database.type.DatabaseType;
 import org.apache.shardingsphere.infra.datasource.pool.creator.DataSourcePoolCreator;
 import org.apache.shardingsphere.infra.datasource.pool.destroyer.DataSourcePoolDestroyer;
@@ -33,22 +32,20 @@ import java.util.Map.Entry;
 /**
  * Data source properties validator.
  */
-@RequiredArgsConstructor
 public final class DataSourcePropertiesValidator {
-    
-    private final DatabaseType databaseType;
     
     /**
      * Validate data source properties map.
      * 
      * @param dataSourcePropertiesMap data source properties map
+     * @param databaseType database type
      * @throws InvalidResourcesException invalid resources exception
      */
-    public void validate(final Map<String, DataSourceProperties> dataSourcePropertiesMap) throws InvalidResourcesException {
+    public void validate(final Map<String, DataSourceProperties> dataSourcePropertiesMap, final DatabaseType databaseType) throws InvalidResourcesException {
         Collection<String> errorMessages = new LinkedList<>();
         for (Entry<String, DataSourceProperties> entry : dataSourcePropertiesMap.entrySet()) {
             try {
-                validate(entry.getKey(), entry.getValue());
+                validate(entry.getKey(), entry.getValue(), databaseType);
             } catch (final InvalidDataSourcePropertiesException ex) {
                 errorMessages.add(ex.getMessage());
             }
@@ -58,11 +55,11 @@ public final class DataSourcePropertiesValidator {
         }
     }
     
-    private void validate(final String dataSourceName, final DataSourceProperties dataSourceProps) throws InvalidDataSourcePropertiesException {
+    private void validate(final String dataSourceName, final DataSourceProperties dataSourceProps, final DatabaseType databaseType) throws InvalidDataSourcePropertiesException {
         DataSource dataSource = null;
         try {
             dataSource = DataSourcePoolCreator.create(dataSourceProps);
-            checkFailFast(dataSource);
+            checkFailFast(dataSource, databaseType);
             // CHECKSTYLE:OFF
         } catch (final Exception ex) {
             // CHECKSTYLE:ON
@@ -74,8 +71,8 @@ public final class DataSourcePropertiesValidator {
         }
     }
     
-    private void checkFailFast(final DataSource dataSource) throws SQLException {
-        if (!dataSource.getConnection().getMetaData().getDatabaseProductName().equals(databaseType.getType().toLowerCase())) {
+    private void checkFailFast(final DataSource dataSource, final DatabaseType databaseType) throws SQLException {
+        if (!dataSource.getConnection().getMetaData().getDatabaseProductName().equals(databaseType.getType())) {
             throw new SQLException("Protocol mismatch for datasource.");
         }
     }

--- a/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesValidator.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesValidator.java
@@ -73,7 +73,7 @@ public final class DataSourcePropertiesValidator {
     
     private void checkFailFast(final DataSource dataSource, final DatabaseType databaseType) throws SQLException {
         if (!dataSource.getConnection().getMetaData().getDatabaseProductName().equals(databaseType.getType())) {
-            throw new SQLException("Protocol mismatch for datasource.");
+            throw new SQLException("Protocol mismatch for data source.");
         }
     }
 }

--- a/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesValidatorTest.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesValidatorTest.java
@@ -18,6 +18,8 @@
 package org.apache.shardingsphere.infra.datasource.props;
 
 import com.zaxxer.hikari.HikariDataSource;
+import org.apache.shardingsphere.infra.database.type.dialect.H2DatabaseType;
+import org.apache.shardingsphere.infra.database.type.dialect.MySQLDatabaseType;
 import org.apache.shardingsphere.infra.distsql.exception.resource.InvalidResourcesException;
 import org.junit.Test;
 
@@ -29,7 +31,12 @@ public final class DataSourcePropertiesValidatorTest {
     
     @Test
     public void assertValidateSuccess() throws InvalidResourcesException {
-        new DataSourcePropertiesValidator().validate(Collections.singletonMap("name", new DataSourceProperties(HikariDataSource.class.getName(), createValidProperties())));
+        new DataSourcePropertiesValidator(new H2DatabaseType()).validate(Collections.singletonMap("name", new DataSourceProperties(HikariDataSource.class.getName(), createValidProperties())));
+    }
+    
+    @Test(expected = InvalidResourcesException.class)
+    public void assertDatabaseTypeInValidateFail() throws InvalidResourcesException {
+        new DataSourcePropertiesValidator(new MySQLDatabaseType()).validate(Collections.singletonMap("name", new DataSourceProperties(HikariDataSource.class.getName(), createValidProperties())));
     }
     
     private Map<String, Object> createValidProperties() {
@@ -43,7 +50,7 @@ public final class DataSourcePropertiesValidatorTest {
     
     @Test(expected = InvalidResourcesException.class)
     public void assertValidateFailed() throws InvalidResourcesException {
-        new DataSourcePropertiesValidator().validate(Collections.singletonMap("name", new DataSourceProperties(HikariDataSource.class.getName(), createInvalidProperties())));
+        new DataSourcePropertiesValidator(new H2DatabaseType()).validate(Collections.singletonMap("name", new DataSourceProperties(HikariDataSource.class.getName(), createInvalidProperties())));
     }
     
     private Map<String, Object> createInvalidProperties() {

--- a/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesValidatorTest.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesValidatorTest.java
@@ -31,12 +31,12 @@ public final class DataSourcePropertiesValidatorTest {
     
     @Test
     public void assertValidateSuccess() throws InvalidResourcesException {
-        new DataSourcePropertiesValidator(new H2DatabaseType()).validate(Collections.singletonMap("name", new DataSourceProperties(HikariDataSource.class.getName(), createValidProperties())));
+        new DataSourcePropertiesValidator().validate(Collections.singletonMap("name", new DataSourceProperties(HikariDataSource.class.getName(), createValidProperties())), new H2DatabaseType());
     }
     
     @Test(expected = InvalidResourcesException.class)
     public void assertDatabaseTypeInValidateFail() throws InvalidResourcesException {
-        new DataSourcePropertiesValidator(new MySQLDatabaseType()).validate(Collections.singletonMap("name", new DataSourceProperties(HikariDataSource.class.getName(), createValidProperties())));
+        new DataSourcePropertiesValidator().validate(Collections.singletonMap("name", new DataSourceProperties(HikariDataSource.class.getName(), createValidProperties())), new MySQLDatabaseType());
     }
     
     private Map<String, Object> createValidProperties() {
@@ -50,7 +50,7 @@ public final class DataSourcePropertiesValidatorTest {
     
     @Test(expected = InvalidResourcesException.class)
     public void assertValidateFailed() throws InvalidResourcesException {
-        new DataSourcePropertiesValidator(new H2DatabaseType()).validate(Collections.singletonMap("name", new DataSourceProperties(HikariDataSource.class.getName(), createInvalidProperties())));
+        new DataSourcePropertiesValidator().validate(Collections.singletonMap("name", new DataSourceProperties(HikariDataSource.class.getName(), createInvalidProperties())), new H2DatabaseType());
     }
     
     private Map<String, Object> createInvalidProperties() {

--- a/shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/ContextManager.java
+++ b/shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/ContextManager.java
@@ -153,7 +153,7 @@ public final class ContextManager implements AutoCloseable {
             Optional.ofNullable(toBeDeletedTableName).ifPresent(optional -> dropTable(databaseName, schemaName, optional));
         }
     }
-
+    
     /**
      * Alter schema.
      *

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/updatable/ImportDatabaseConfigurationHandler.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/updatable/ImportDatabaseConfigurationHandler.java
@@ -71,7 +71,7 @@ import java.util.Map.Entry;
  */
 public final class ImportDatabaseConfigurationHandler extends UpdatableRALBackendHandler<ImportDatabaseConfigurationStatement> {
     
-    private final DataSourcePropertiesValidator validator = new DataSourcePropertiesValidator(getConnectionSession().getDatabaseType());
+    private final DataSourcePropertiesValidator validator = new DataSourcePropertiesValidator();
     
     private final ShardingRuleConfigurationImportChecker shardingRuleConfigurationImportChecker = new ShardingRuleConfigurationImportChecker();
     
@@ -131,7 +131,7 @@ public final class ImportDatabaseConfigurationHandler extends UpdatableRALBacken
         for (Entry<String, YamlProxyDataSourceConfiguration> entry : yamlDataSourceMap.entrySet()) {
             dataSourcePropsMap.put(entry.getKey(), DataSourcePropertiesCreator.create(HikariDataSource.class.getName(), dataSourceConfigSwapper.swap(entry.getValue())));
         }
-        validator.validate(dataSourcePropsMap);
+        validator.validate(dataSourcePropsMap, getConnectionSession().getDatabaseType());
         try {
             ProxyContext.getInstance().getContextManager().updateResources(databaseName, dataSourcePropsMap);
         } catch (final SQLException ex) {

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/updatable/ImportDatabaseConfigurationHandler.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/updatable/ImportDatabaseConfigurationHandler.java
@@ -71,7 +71,7 @@ import java.util.Map.Entry;
  */
 public final class ImportDatabaseConfigurationHandler extends UpdatableRALBackendHandler<ImportDatabaseConfigurationStatement> {
     
-    private final DataSourcePropertiesValidator validator = new DataSourcePropertiesValidator();
+    private final DataSourcePropertiesValidator validator = new DataSourcePropertiesValidator(getConnectionSession().getDatabaseType());
     
     private final ShardingRuleConfigurationImportChecker shardingRuleConfigurationImportChecker = new ShardingRuleConfigurationImportChecker();
     

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/resource/AddResourceBackendHandler.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/resource/AddResourceBackendHandler.java
@@ -53,14 +53,14 @@ public final class AddResourceBackendHandler extends DatabaseRequiredBackendHand
     public AddResourceBackendHandler(final AddResourceStatement sqlStatement, final ConnectionSession connectionSession) {
         super(sqlStatement, connectionSession);
         databaseType = connectionSession.getDatabaseType();
-        validator = new DataSourcePropertiesValidator(connectionSession.getDatabaseType());
+        validator = new DataSourcePropertiesValidator();
     }
     
     @Override
     public ResponseHeader execute(final String databaseName, final AddResourceStatement sqlStatement) throws DistSQLException {
         checkSQLStatement(databaseName, sqlStatement);
         Map<String, DataSourceProperties> dataSourcePropsMap = ResourceSegmentsConverter.convert(databaseType, sqlStatement.getDataSources());
-        validator.validate(dataSourcePropsMap);
+        validator.validate(dataSourcePropsMap, databaseType);
         try {
             ProxyContext.getInstance().getContextManager().updateResources(databaseName, dataSourcePropsMap);
         } catch (final SQLException | ShardingSphereException ex) {

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/resource/AddResourceBackendHandler.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/resource/AddResourceBackendHandler.java
@@ -53,7 +53,7 @@ public final class AddResourceBackendHandler extends DatabaseRequiredBackendHand
     public AddResourceBackendHandler(final AddResourceStatement sqlStatement, final ConnectionSession connectionSession) {
         super(sqlStatement, connectionSession);
         databaseType = connectionSession.getDatabaseType();
-        validator = new DataSourcePropertiesValidator();
+        validator = new DataSourcePropertiesValidator(connectionSession.getDatabaseType());
     }
     
     @Override

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/resource/AlterResourceBackendHandler.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/resource/AlterResourceBackendHandler.java
@@ -60,14 +60,14 @@ public final class AlterResourceBackendHandler extends DatabaseRequiredBackendHa
     public AlterResourceBackendHandler(final AlterResourceStatement sqlStatement, final ConnectionSession connectionSession) {
         super(sqlStatement, connectionSession);
         databaseType = connectionSession.getDatabaseType();
-        validator = new DataSourcePropertiesValidator(connectionSession.getDatabaseType());
+        validator = new DataSourcePropertiesValidator();
     }
     
     @Override
     public ResponseHeader execute(final String databaseName, final AlterResourceStatement sqlStatement) throws DistSQLException {
         checkSQLStatement(databaseName, sqlStatement);
         Map<String, DataSourceProperties> dataSourcePropsMap = ResourceSegmentsConverter.convert(databaseType, sqlStatement.getDataSources());
-        validator.validate(dataSourcePropsMap);
+        validator.validate(dataSourcePropsMap, databaseType);
         try {
             ProxyContext.getInstance().getContextManager().updateResources(databaseName, dataSourcePropsMap);
         } catch (final SQLException | ShardingSphereException ex) {

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/resource/AlterResourceBackendHandler.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/resource/AlterResourceBackendHandler.java
@@ -60,7 +60,7 @@ public final class AlterResourceBackendHandler extends DatabaseRequiredBackendHa
     public AlterResourceBackendHandler(final AlterResourceStatement sqlStatement, final ConnectionSession connectionSession) {
         super(sqlStatement, connectionSession);
         databaseType = connectionSession.getDatabaseType();
-        validator = new DataSourcePropertiesValidator();
+        validator = new DataSourcePropertiesValidator(connectionSession.getDatabaseType());
     }
     
     @Override

